motr/ut/idx_dix.c

```diff
@@ -782,15 +782,282 @@ struct m0_client* st_get_instance()
 	return ut_m0c;
 }
 
+#include "dtm0/helper.h"
+#include "dtm0/service.h"
+
+struct dtm0_ut_ctx {
+	struct m0_fid           duc_cli_svc_fid;
+	struct m0_fid           duc_srv_svc_fid;
+	struct m0_reqh_service *duc_srv_svc;
+	struct m0_idx           duc_idx;
+	struct m0_container     duc_realm;
+	struct m0_fid           duc_ifid;
+};
+
+static struct dtm0_ut_ctx duc = {};
+
+static int duc_setup(void)
+{
+	const char              *cl_ep_addr   = "0@lo:12345:34:2";
+	struct m0_fid            cli_srv_fid  = M0_FID_INIT(0x7300000000000001,
+							    0x1a);
+	struct m0_fid            srv_dtm0_fid = M0_FID_INIT(0x7300000000000001,
+							    0x1c);
+	struct m0_reqh_service  *srv_srv = NULL;
+	struct m0_reqh          *srv_reqh;
+	int                      rc;
+	struct m0_container     *realm = &duc.duc_realm;
+
+	m0_fi_enable("m0_dtm0_in_ut", "ut");
+	rc = ut_suite_mt_idx_dix_init();
+	M0_UT_ASSERT(rc == 0);
+
+	m0_container_init(realm, NULL, &M0_UBER_REALM, ut_m0c);
+
+	if (ENABLE_DTM0) {
+		/* Connect to the server */
+		M0_UT_ASSERT(ut_m0c->m0c_dtms != NULL);
+		srv_reqh = &dix_ut_sctx.rsx_motr_ctx.cc_reqh_ctx.rc_reqh;
+		srv_srv = m0_reqh_service_lookup(srv_reqh, &srv_dtm0_fid);
+		rc = m0_dtm0_service_process_connect(srv_srv, &cli_srv_fid,
+						     cl_ep_addr, false);
+		M0_UT_ASSERT(rc == 0);
+	}
+
+	general_ifid_fill(&duc.duc_ifid, true);
+
+	/* Save the context */
+	duc.duc_cli_svc_fid = cli_srv_fid;
+	duc.duc_srv_svc_fid = srv_dtm0_fid;
+	duc.duc_srv_svc = srv_srv;
+
+	return 0;
+}
+
+static void idx_setup(void)
+{
+	struct m0_op            *op = NULL;
```

Comment:
such nullification is a sign of unexpected side effect inside the use of this structure. either put a comment why it's needed or avoid = NULL.